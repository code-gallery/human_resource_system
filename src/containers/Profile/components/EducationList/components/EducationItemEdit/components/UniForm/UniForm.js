import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'
import { renderValidation } from 'containers/Profile/utils/validation'
import CustomDatePicker from 'components/FormElements/DatePicker'
// import PDFReader from 'components/FormElements/PDFReader'

class UniForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      degreeSelected: ''
    }
  }

  componentWillMount() {
    this.reader = new FileReader()
    this.reader.addEventListener('load', this.onFileLoad)
  }

  componentWillUnmount() {
    this.reader.removeEventListener('load', this.onFileLoad)
  }

  handleChange = (value, title) => {
    this.props.onFieldChange(title, value)
  }

  onTextChange = (title, event) => {
    this.props.onFieldChange(title, event.target.value)
  }

  filterAwards = (type) => {
    if (type) {
      return this.props.awards.filter((item) => {
        return item.value1 === type
      })
    }
    return this.props.awards
  }

  onFileLoad = () => {
    this.props.onFieldChange('transcript', this.reader.result)
  }

  handleFileChange = (event) => {
    const file = event.target.files[0]
    if (file) {
      this.reader.readAsDataURL(file)
    }
  }

  render() {
    const {
      start_date,
      end_date,
      student_number,
      studied,
      degree,
      grade,
      qualifications,
      invalidFields,
      invalidDates,
      invalidMessage,
      // transcript
    } = this.props
    return (
      <div>
        <div className="form-group">
          <label className={renderValidation(invalidFields, 'studied')}>Studied</label>
          <input
            name="studied"
            type="text"
            value={studied}
            onChange={v => this.onTextChange('studied', v)}
          />
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label className={renderValidation(invalidFields, 'degree')}>Degree/Qualifications</label>
              <Select
                name="select-degree"
                options={qualifications}
                labelKey="value2"
                valueKey="value2"
                value={{ value2: degree || '' }}
                onChange={v => {
                  this.handleChange(v ? v.value2 : '', 'degree')
                  this.setState({ degreeSelected: v.value1 }, () => {
                    this.handleChange('', 'grade')
                  })
                }}
              />
            </div>
          </div>
          {this.state.degreeSelected && (
            <div className="col-xs-12 col-md-6">
              <div className="form-group">
                <label>Award</label>
                <Select
                  name="select-content"
                  options={this.filterAwards(this.state.degreeSelected)}
                  labelKey="value2"
                  valueKey="value2"
                  value={{ value2: grade || '' }}
                  onChange={v => this.handleChange(v ? v.value2 : '', 'grade')}
                />
              </div>
            </div>
          )}
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <CustomDatePicker
              handleDate={this.handleChange}
              date={start_date}
              invalid={invalidFields.indexOf('start_date') !== -1}
              title="start_date"
              label="Start Date" />
          </div>
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <CustomDatePicker
                handleDate={this.handleChange}
                date={end_date}
                invalid={invalidFields.indexOf('end_date') !== -1}
                title="end_date"
                label="End Date"
                endDate />
            </div>
          </div>
        </div>
        {invalidDates && (<div className="required-error">{invalidMessage}</div>)}
        <div className="form-group">
          <p className="help-text">Adding your student number may speed up verification.</p>
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label>Student Number (Optional)</label>
              <input
                name="student-number"
                type="text"
                value={student_number}
                onChange={(data) => this.handleChange(data.target.value, 'student_number')}
              />
            </div>
          </div>
          { /* <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label>&nbsp;</label>
              <div className="btn-custom-file-upload">
                <label>
                  <input
                    type="file"
                    onChange={this.handleFileChange}
                    accept="application/pdf"
                  />
                  <span>Upload Transcript (Optional)</span>
                </label>
              </div>
            </div>
          </div>
          <div className="row">
          transcript && (
            <PDFReader transcript={ /* transcript} />
          )
          </div>*/ }
        </div>
      </div>
    )
  }
}

UniForm.propTypes = {
  onFieldChange: PropTypes.func,
  student_number: PropTypes.string,
  studied: PropTypes.string,
  start_date: PropTypes.string,
  end_date: PropTypes.string,
  degree: PropTypes.string,
  grade: PropTypes.string,
  invalidFields: PropTypes.array,
  awards: PropTypes.array,
  qualifications: PropTypes.array,
  invalidDates: PropTypes.bool,
  invalidMessage: PropTypes.string,
  transcript: PropTypes.string
}

export default UniForm
