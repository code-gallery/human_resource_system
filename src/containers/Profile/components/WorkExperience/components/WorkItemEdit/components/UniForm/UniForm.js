import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'
import _isNil from 'lodash/isNil'
import Autocomplete from 'react-google-autocomplete'
import { renderValidation } from 'containers/Profile/utils/validation'
import CustomDatePicker from 'components/FormElements/DatePicker'
import BaseForm from 'containers/Profile/components/AwardList/components/AwardItemEdit/components/Forms/BaseForm'
import WorkExperienceAchievement from 'containers/Profile/components/WorkExperienceAchievement'

export default class UniForm extends BaseForm {
  render() {
    const {
      position,
      location,
      achievements,
      start_date,
      end_date,
      industry,
      industries,
      invalidFields,
      invalidDates,
      invalidMessage,
      onFieldChange,
      employment_type
    } = this.props
    const locationPlaceholder = (_isNil(location) || location === '') ?
      'Enter a location' : location

    return (
      <div>
        <div className="form-group">
          <label className={renderValidation(invalidFields, 'position')}>Position</label>
          <input type="text" name="position" value={position || ''} onChange={v => this.onTextChange('position', v)} />
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label>Location</label>
              <Autocomplete
                onPlaceSelected={res => this.onTextChange('location', res)}
                placeholder={locationPlaceholder}
                types={[ '(cities)' ]}
              />
            </div>
          </div>
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label className={renderValidation(invalidFields, 'industry')}>Industry</label>
              <Select
                name="select-content"
                options={industries}
                labelKey="text"
                valueKey="text"
                value={{ text: industry || '' }}
                onChange={v => this.handleChange(v ? v.text : '', 'industry')}
              />
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <CustomDatePicker
                handleDate={this.handleChange}
                invalid={invalidFields.indexOf('start_date') !== -1}
                date={start_date}
                title="start_date"
                label="Start Date"
              />
            </div>
          </div>
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <CustomDatePicker
                handleDate={this.handleChange}
                invalid={invalidFields.indexOf('end_date') !== -1}
                date={end_date}
                title="end_date"
                label="End Date"
                endDate
              />
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label className={renderValidation(invalidFields, 'employment_type')}>Employment Type</label>
              <Select
                name="select-content"
                options={[
                  {
                    value: 'permanent',
                    label: 'Permanent Employee'
                  },
                  {
                    value: 'contractor',
                    label: 'Contractor'
                  }
                ]}
                value={employment_type}
                onChange={v => this.handleChange(v ? v.value : '', 'employment_type')}
              />
            </div>
          </div>
        </div>
        <WorkExperienceAchievement achievements={achievements} onFieldChange={onFieldChange} />
        {invalidDates && (<div className="required-error">{invalidMessage}</div>)}
      </div>
    )
  }
}

UniForm.propTypes = {
  onFieldChange: PropTypes.func.isRequired,
  achievements: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object
  ]),
  position: PropTypes.string,
  location: PropTypes.string,
  start_date: PropTypes.string,
  end_date: PropTypes.string,
  industry: PropTypes.string,
  industries: PropTypes.array,
  invalidFields: PropTypes.array,
  invalidDates: PropTypes.bool,
  invalidMessage: PropTypes.string,
  employment_type: PropTypes.string
}
