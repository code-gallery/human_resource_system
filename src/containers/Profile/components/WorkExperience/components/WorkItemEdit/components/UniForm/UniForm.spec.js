import React from 'react'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import UniForm from './index'

configure({adapter: new Adapter()});
describe('containers/Profile/components/.../UniForm', () => {
  const commonProps = {
    deleteWork: jest.fn(),
    saveEntity:jest.fn(),
    toggleEdit:jest.fn(),
    toggleBlockEditing:jest.fn(),
    onFieldChange: jest.fn(),
    orgRegistered: jest.fn(),
    invalidFields: [],
    invalidMessage: 'Error occured',
    invalidDates: false,
    industries: [
      { text: 'Accounting' },
      { text: 'Animation' },
      { text: 'IT' }
    ]
  }

  describe('when all inputs have no values', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps
      }
      const tree = renderer.create(
        <UniForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when invalidDates is true', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        invalidDates: true
      }
      const tree = renderer.create(
        <UniForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when "position" and "start_date" are invalid', () => {
    const props = {
      ...commonProps,
      position: '',
      location: 'London, UK',
      industry: 'Accounting',
      end_date: '2016-03-01',
      start_date: '',
      organisation: 'City of Oxford College',
      invalidFields: [ 'position', 'start_date' ]
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <UniForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when all inputs have values', () => {
    const props = {
      ...commonProps,
      position: 'Developer',
      location: 'London, UK',
      industry: 'Accounting',
      end_date: '2016-06-01',
      start_date: '2016-03-01',
      organisation: 'City of Oxford College'
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <UniForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    describe('handleChange', () => {
      it('calls props.onFieldChange correctly', () => {
        const wrapper = shallow(
          <UniForm {...props} />
        )
        props.onFieldChange.mockClear()
        wrapper.instance().handleChange('value', 'field')
        expect(props.onFieldChange)
          .toHaveBeenCalledWith('field', 'value')
      })
    })

    describe('onTextChange', () => {
      describe('when "position" changes', () => {
        it('calls props.onFieldChange correctly', () => {
          const wrapper = shallow(
            <UniForm {...props} />
          )
          const event = { target: { name: 'position', value: 'Developer' } }
          props.onFieldChange.mockClear()
          wrapper.find('input[name="position"]').simulate('change', event)
          expect(props.onFieldChange)
            .toHaveBeenCalledWith('position', 'Developer')
        })
      })
    })
  })
})
