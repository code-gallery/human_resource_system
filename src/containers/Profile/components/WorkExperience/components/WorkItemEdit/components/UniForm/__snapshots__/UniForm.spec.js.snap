// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/.../UniForm when "position" and "start_date" are invalid renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className="required-error"
    >
      Position
    </label>
    <input
      name="position"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <input
          placeholder="London, UK"
        />
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Industry
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value="Accounting"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-6--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-6--value-item"
                  role="option"
                >
                  Accounting
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-6--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className="required-error"
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              April 2018
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              March 2016
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-7--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-7--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="WorkExperienceAchievement"
  >
    <div
      className="form-group"
    >
      <p
        className="help-text"
      >
        You can add 5 achievements
      </p>
    </div>
    <div
      className="form-group"
    >
      <button
        className="add-row-btn"
        onClick={[Function]}
      >
        Add achievement
      </button>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../UniForm when all inputs have no values renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Position
    </label>
    <input
      name="position"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <input
          placeholder="Enter a location"
        />
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Industry
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value=""
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-2--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-2--value-item"
                  role="option"
                >
                  
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-2--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              April 2018
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              Present
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-3--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-3--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="WorkExperienceAchievement"
  >
    <div
      className="form-group"
    >
      <p
        className="help-text"
      >
        You can add 5 achievements
      </p>
    </div>
    <div
      className="form-group"
    >
      <button
        className="add-row-btn"
        onClick={[Function]}
      >
        Add achievement
      </button>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../UniForm when all inputs have values renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Position
    </label>
    <input
      name="position"
      onChange={[Function]}
      type="text"
      value="Developer"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <input
          placeholder="London, UK"
        />
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Industry
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value="Accounting"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-8--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-8--value-item"
                  role="option"
                >
                  Accounting
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-8--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              March 2016
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              June 2016
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-9--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-9--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="WorkExperienceAchievement"
  >
    <div
      className="form-group"
    >
      <p
        className="help-text"
      >
        You can add 5 achievements
      </p>
    </div>
    <div
      className="form-group"
    >
      <button
        className="add-row-btn"
        onClick={[Function]}
      >
        Add achievement
      </button>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../UniForm when invalidDates is true renders correctly 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Position
    </label>
    <input
      name="position"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Location
        </label>
        <input
          placeholder="Enter a location"
        />
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Industry
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value=""
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-4--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-4--value-item"
                  role="option"
                >
                  
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-4--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              April 2018
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              Present
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-5--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-5--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="WorkExperienceAchievement"
  >
    <div
      className="form-group"
    >
      <p
        className="help-text"
      >
        You can add 5 achievements
      </p>
    </div>
    <div
      className="form-group"
    >
      <button
        className="add-row-btn"
        onClick={[Function]}
      >
        Add achievement
      </button>
    </div>
  </div>
  <div
    className="required-error"
  >
    Error occured
  </div>
</div>
`;
